src/components/Modal/Modal.jsx

```diff
@@ -41,6 +42,17 @@ class Modal extends React.Component {
     if (this.props.clickToClose) {
       document.addEventListener('click', this.handleDocumentClicked, true);
     }
+    // Conditionally track the event.
+    if (this.props.enableAnalytics) {
+      dispatchAnalyticsEvent({
+        'type': 'Modal',
+        'action': 'show',
+        'status': this.props.status,
+        'title': this.props.title,
+        'primaryButtonText': this.props.primaryButton?.text,
+        'seondaryButtonText': this.props.secondaryButton?.text,
```

Comment:
I'll have more opinions about the shape of this when we talk later, but a few things until then:
- You don't need to quote your object property names
  - If you merge the latest `master` branch in here, it'll set up commit hooks to auto-format and lint your code when you go to commit :sparkles: 
- Looks like you've got a typo for `secondaryButtonText`

Reply:
Do we need to add the `mocha` plugin and env to `.eslintrc.json`?  I'm seeing a lot of linting errors on the global vars mocha uses to run tests (_describe_, _it_, etc)

Reply:
Yeah, good catch.

For posterity, @bsmartin-ep made this addition in [this separate PR](https://github.com/department-of-veterans-affairs/component-library/pull/38).

Reply:
Done.